cpp/src/slice2java/Gen.cpp

```diff
@@ -2343,13 +2296,13 @@ Slice::Gen::TypesVisitor::visitClassDefStart(const ClassDefPtr& p)
                 //
                 // Generate a constructor accepting parameters for just the required members.
                 //
-                out << sp << nl << "public " << fixKwd(name) << spar;
+                out << sp << nl << "public " << name << spar;
                 vector<string> parameters;
                 for (const auto& allDataMember : allDataMembers)
```

Comment:
not related to the PR, but `allDataMember` seems an awful name for the iteration variable.